exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/Table.kt

```diff
@@ -1723,6 +1729,34 @@ open class Table(name: String = "") : ColumnSet(), DdlAware {
         return createAutoIncColumnSequence() + createTable + createConstraint
     }
 
+    private fun addLongColumnCheckConstraintIfNeeded() {
+        if (currentDialect is OracleDialect || currentDialect is SQLiteDialect) {
+            columns.filter { it.columnType is LongColumnType }.forEach { column ->
+                val name = column.name
+                val checkName = "${generatedSignedCheckPrefix}long_$name"
+                if (checkConstraints.none { it.first == checkName }) {
+                    column.check(checkName) {
```

Comment:
I don't know if it's ok or not, but to me it looks a bit suspicious at the current moment.

We actually modify the `table` object, so user will have changes in table object if he tries to create statement for that table.

The method `createStatement()` does not look like the one that should modify table object, so some users may be surprised

Reply:
I agree. I didn't like this solution, to be frank, but I didn't have any other ideas. If you have any suggestions on how to do this in a cleaner way, please let me know.

Reply:
It's a good question, I'm not sure about what is the behavior we want to achieve.

If we want to guarantee that only long values would be stored in the database base and exception would be thrown otherwise, it could be problematic (like in the PR)

I tried to check SQLite, and I found out that if the column type `INTEGER` is used, then the column will store only Kotlin's Long values. But it would not complain if the user tried to save the value beyond `Long.MIN_VALUE`/`Long.MAX_VALUE`, it will just "cut" the values to the range.

Reply:
@e5l What do you think?

Reply:
I agree with @obabichevjb about modifying tables in `createStatement`, so I'm good if we're not doing that.

The one major thing to consider is a strict mode (like https://dev.mysql.com/doc/refman/8.4/en/sql-mode.html) or strict tables (https://www.sqlite.org/stricttables.html). We should mention these in the docs and check if we're working as expected if those are enabled.